packages/simcore-sdk/src/simcore_sdk/node_data/data_manager.py

```diff
@@ -66,36 +66,51 @@ async def push(
         return await _push_file(user_id, project_id, node_uuid, archive_file_path, None)
 
 
-async def _pull_file(user_id: int, project_id: str, node_uuid: str, file_path: Path):
+async def _pull_file(
+    user_id: int,
+    project_id: str,
+    node_uuid: str,
+    file_path: Path,
+    save_to: Optional[Path] = None,
+):
+    destination_path = file_path if save_to is None else save_to

```

Comment:
TIP: i personally like this approach. It is less verbose. But yours is perfectly fine.
```python
destination_path = save_to or file_path
```

Reply:
Be careful with that one, as values which are not None may also get evaluated to "false" and can break (like empty strings, empty lists, etc...). In this case your proposal is valid.